conans/model/package_metadata.py

```diff
@@ -45,15 +43,15 @@ def revision(self):
 
     @revision.setter
     def revision(self, r):
-        self._revision = DEFAULT_REVISION_V1 if r is None else r
+        self._revision = r
```

Comment:
A ``assert r is not None`` here is the typical thing that can be useful to spot errors quickly